willow/plugins/pillow.py

```diff
@@ -227,19 +278,32 @@ def save_as_png(self, f, optimize: bool = False, apply_optimizers: bool = True):
         :param apply_optimizers: controls whether to run any configured optimizer libraries
         :return: PNGImageFile
         """
+
+        kwargs = {}
+        image = self.image
+        icc_profile = self.get_icc_profile()
+        if icc_profile is not None:
+            # If the image is in CMYK mode *and* has an ICC profile, we need to be more diligent
+            # about how we handle the color conversion to RGB. We don't want to retain
+            # the color profile as-is because it is not meant for RGB images and
+            # will result in inaccurate colors. The transformation to sRGB should result
+            # in a more accurate representation of the original image, though
+            # it will likely not be perfect.
+            if self.image.mode == "CMYK":
+                pillow_image = self.transform_colorspace_to_srgb()
+                image = pillow_image.image
+                kwargs["icc_profile"] = pillow_image.get_icc_profile()
+            else:
+                kwargs["icc_profile"] = icc_profile
+
         if self.image.mode == "CMYK":
             image = self.image.convert("RGB")
         else:
             image = self.image
```

Comment:
This now does double duty. I believe this now need to be

```suggestion
        elif self.image.mode == "CMYK":
            # we don't have an ICC profile, and the image is in CMYK, so convert to RGB
            image = self.image.convert("RGB")
```

Reply:
Little oversight of mine, thanks!